### ISSUE TITLE
Proposal: Reorganise displayers and plotters

### ISSUE THREAD

**Issue by DanShort12** [Fusion-Power-Plant-Framework/bluemira#531]
## Description of issue / requirement to address

The various displayers and plotters currently reside in the `display` module. This often leads to circular imports because the functionality to retrieve the plotter e.g. `_get_plotter_class` needs to know about the classes that can be plotted/displayed. This inter-dependency suggests that in reality the plotter/displayer should be in the same module as the thing that it can plot / display.

## Proposed solution

Move the specific plotters and displayers into the same modules as the things that are being plotted/displayed. It should also be possible to register the class to be plotted / displayed with its corresponding plotter without hard coding them into `_get_plotter_class` and equivalents. This could be done with a decorator to explicitly say what plotter should be used for the class being registered.

## Alternative solutions

We could probably achieve something similar by using `__init_subclass__` and directly assigning the plotter / displayer class to the class to be plotted/displayed.

## Additional context

This will likely leave the `display` module as even smaller than it currently is. It may then be worth considering moving this into `base` to reduce the number of separate modules that we have. However, as `displayer.py` depends on an external code (cadapi) this could be bad for future independence of core modules from external codes (although so much will depend on the cadapi that perhaps this is less of a concern).

**Comment by je-cook** [comment]
I agree the plotter should be close to the thing being plotted (at least in the same module). I havent got any immediate ideas about how to do it and reduce circular imports though...

**Comment by CoronelBuendia** [comment]
Fundamentally, all we are displaying in the `display` module at the moment is geometry. We're certainly not going to be displaying any CAD that does not come from geometry.

In terms of plotting, we could in future display contours for 2-D scalar fields and the like, and other general results, but at the risk of ending up wrapping matplotlib and of endlessly trying to standardise different plots (where sometimes one approach looks better than the other) is high, in my opinion. I'm also wondering (after the fact) why `geometry` and `display` are even different modules. Note that a whole bunch of other modules (e.g. equilibria, magnetostatics, balance of plant) handle their plotting internally to the module. Philosophically we went for being able to "replace" any module with another one, and the corollary, that any module be as standalone as possible. Personally, I consider display and plotting to be a fundamental part of what each module does. From this standpoint, I no longer see why `display` is separate from `geometry`, seeing as the former exclusively serves the latter.
